### DISCUSSION TITLE
Why can everything be null?

### DISCUSSION THREAD

**Discussion by abdonrd** [0no-co/gql.tada#161] [Q&A] [answered]
The readme has this image example:

<img width="100%" alt="Code Editor showing GraphQL queries being edited with gql.tada and GraphLSP" src="https://github.com/0no-co/gql.tada/blob/277ce424a747522ef2ca0d398b113f4f285eb595/website/public/demo-code.png?raw=true" />

And you can see how the data data is typed as:

```typescript
{
  pokemons: ({ id: string | number } | null)[] | null;
}
```

Why is not like this?

```typescript
{
  pokemons: ({ id: string | number })[];
}
```

When I do the query, I always get an array.
It can be empty, but it is always an array.

On what occasion can null arrive? Or am I missing something?

I ask because then the way to use it is:

```typescript
data?.pokemons?.map(pokemon => pokemon!.id)
```

Instead of:

```typescript
data.pokemons.map(pokemon => pokemon.id)
```

Is there any other workaround to have a better development experience?

Thanks!

**Comment by JoviDeCroock** [comment, marked as answer]
I mean, in GraphQL the server sais whether something can be nullable or not. Indicating something as nullable acts like a boundary for errors,... it can also be used on the client to indicate where partial data is allowed. I highly encourage you to read the spec to know more about this.

Basically this isn't a question that can be answered from this library as it's a building block of GraphQL as a technology.

Wrote a bit about it here https://stellate.co/blog/graphql-types-as-a-superpower and a reddit post https://www.reddit.com/r/graphql/s/0bESQWSZ8P

Tldr: when a field errors for a given pokemon that would bubble up to the first non nullable boundary. If we ask for a list of pokemons and then for one pokemon but with more fields the nullability of those fields can instruct us to show partial data.
